custom/logistics/tasks.py

```diff
@@ -101,3 +113,98 @@ def add_products_to_loc(api):
+
+            # set the checkpoint right before the data we are about to process
+            save_stock_data_checkpoint(checkpoint,
+                                       'stock_transaction',
+                                       meta.get('limit') or limit,
+                                       meta.get('offset') or offset,
+                                       date, facility, True)
+            transactions_to_add = []
+            with transaction.commit_on_success():
+                for stocktransaction in stocktransactions:
+                    params = dict(
+                        form_id=xform._id,
+                        date=force_to_datetime(stocktransaction.date),
+                        type='balance',
+                        domain=domain,
+                    )
+                    try:
+                        report, _ = StockReport.objects.get_or_create(**params)
+                    except StockReport.MultipleObjectsReturned:
+                        # legacy
+                        report = StockReport.objects.filter(**params)[0]
+                    try:
+                        sql_product = SQLProduct.objects.get(code=stocktransaction.product, domain=domain)
+                    except SQLProduct.DoesNotExist:
+                        # todo: kkrampa what's the deal with this logic? this should never be true
+                        continue
+                    if stocktransaction.quantity != 0:
+                        transactions_to_add.append(StockTransaction(
+                            case_id=case._id,
+                            product_id=sql_product.product_id,
+                            sql_product=sql_product,
+                            section_id=section_id,
+                            type='receipts' if stocktransaction.quantity > 0 else 'consumption',
+                            stock_on_hand=Decimal(stocktransaction.ending_balance),
+                            quantity=Decimal(stocktransaction.quantity),
+                            report=report
+                        ))
+                    transactions_to_add.append(StockTransaction(
+                        case_id=case._id,
+                        product_id=sql_product.product_id,
+                        sql_product=sql_product,
+                        section_id=section_id,
+                        type='stockonhand',
+                        stock_on_hand=Decimal(stocktransaction.ending_balance),
+                        report=report
+                    ))
+                    products_saved.add(sql_product.product_id)
+
+            if transactions_to_add:
+                # Doesn't send signal
+                StockTransaction.objects.bulk_create(transactions_to_add)
+
+            if not meta.get('next', False):
+                has_next = False
+            else:
+                next_url = meta['next'].split('?')[1]
```

Comment:
don't have much context on this, but urls don't always have a `?` in them so this could throw an error. not sure if this is never the case in this situation though